modules/shortcodes/youtube.php

```diff
@@ -246,16 +286,54 @@ function youtube_id( $url ) {
 	 * @param bool false Enable autoplay for YouTube videos.
 	 */
 	if ( apply_filters( 'jetpack_youtube_allow_autoplay', false ) && isset( $args['autoplay'] ) ) {
-		$autoplay = '&autoplay=' . (int) $args['autoplay'];
+		$params['autoplay'] = (int) $args['autoplay'];
 	}
 
-	if (
-		( isset( $url['path'] ) && '/videoseries' === $url['path'] )
```

Comment:
> Since we're touching this, let's move to type casting, [just like in Core](https://core.trac.wordpress.org/changeset/49108)?

Here?

```diff
--- a/modules/shortcodes/youtube.php
+++ b/modules/shortcodes/youtube.php
@@ -215,7 +215,7 @@ function youtube_id( $url ) {
 	if ( ! isset( $args['autohide'] ) || ( $args['autohide'] < 0 || 2 < $args['autohide'] ) ) {
 		$params['autohide'] = 2;
 	} else {
-		$params['autohide'] = absint( $args['autohide'] );
+		$params['autohide'] = (int) $args['autohide'];
 	}
 
 	$start = 0;
```

Reply:
As far as I can tell, a `videoseries` is just an alias for a playlist.

So given this URL: https://www.youtube.com/videoseries?list=PL56C3506BBE979C1B

Previously this would result in an iframe with the `src`: 

https://www.youtube.com/embed/videoseries?list=PL56C3506BBE979C1B&hl=en_US

But notice the `list` parameter. With this PR, this instead gets output as an iframe with the `src`:

https://www.youtube.com/embed?version=3&rel=1&showsearch=0&showinfo=1&iv_load_policy=1&fs=1&autohide=2&wmode=transparent&listType=playlist&list=PL56C3506BBE979C1B

Comparing these two pages, they are identical. The only thing that matters is the `list` param, no?